lib/std/json.zig

```diff
@@ -1624,11 +1691,16 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                 }
             }
 
+            comptime const field_aliases = buildFieldAliases(T);
+
             while (true) {
                 switch ((try tokens.next()) orelse return error.UnexpectedEndOfJson) {
                     .ObjectEnd => break,
                     .String => |stringToken| {
-                        const key_source_slice = stringToken.slice(tokens.slice, tokens.i - 1);
+                        const key_source_slice = findFieldAlias(
+                            field_aliases,
+                            stringToken.slice(tokens.slice, tokens.i - 1),
```

Comment:
This isn't necessarily what you think it is: it might have escapes. Consult `stringToken.escapes` to know.

Reply:
I'll look into adding escape support soon.  @daurnimator so put this into ParseOptions?  How does this look?: 
```zig
const S = struct {value: u8};
const s = try parse(
    S, 
    &TokenStream.init("{\"__value\": 42}"), 
    .{.aliases = .{.value = "__value"}},
);
```
I'm a little worried this is less re-usable and would promode copy-pasting.  But maybe thats necessary if you wanted to parse different json data into the same struct.  I guess it would be easy enough to store the alias info in the struct decl again like so:
```zig
const S = struct {
    value: u8,
    pub const aliases = .{.value = "__value"};
};
const s = try parse(
    S, 
    &TokenStream.init("{\"__value\": 42}"), 
    .{.aliases = S.aliases}
);
```
Thoughts?